plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/TemplateViewerFilter.java

```diff
@@ -38,15 +38,19 @@ public boolean select(Viewer viewer, Object parentElement, Object element) {
 		if(!(element instanceof IResource)) {
 			return false;
 		}
+		final String text = filterText.getText();
+		if(StringUtils.isBlank(text)) {
+			return true;
+		}
 		IResource resource = (IResource) element;
-		if(StringUtils.isBlank(filterText.getText())) {
+		if(resource.getName().contains(text)) {
```

Comment:
I'd agree on the logical requirement where resources would need to have a resource but I could still imagine an error in unmarshalling where the name doesnt get extracted from json. Then dmr catches in, which is fine for our current situation but imagine we'd change the unmarshalling technology and all of a sudden one would unexpectly get an NPE here. I'd personnally add a simple null-check here jsut to make it more robust even if not strictly required.